packages/material-ui/src/utils/renderProps.js

```diff
@@ -0,0 +1,29 @@
+// this is basically copied from react-powerplug:
+// https://github.com/renatorib/react-powerplug/blob/master/src/utils/renderProps.js
+
+import warning from 'warning';
+
+const isFn = prop => typeof prop === 'function';
+
+/**
+ * renderProps
+ * is a render/children props interop.
+ * will pick up the prop that was used,
+ * or children if both are used
+ */
+
+const renderProps = ({ children, render }, ...props) => {
```

Comment:
I sure wouldn't mind removing it, I only put it in there because some libs like `react-powerplug` seem to include it to appease people who think passing a child function is some kind of blasphemy

Reply:
Well, I'm happy about being as opinionated as React.createContext API.

Reply:
Oh that's true.  I'm glad the React folks decided to do that, because there were some really dogmatic people complaining that `children` should always be a `React.Node`.

It's about the same as how in all of the Angular vs. React vs. Vue type videos I've seen, people said they didn't like React because something about putting HTML in JavaScript didn't feel right to them.  They didn't have any more substantive criticism than that.